### ISSUE TITLE
Is there any way to concatenate two TabularDatasets?

### ISSUE THREAD

**Issue by ben0it8** [pytorch/text#375]
Hello,
 
Let's say I've split my train set to train and val and I also have a test set.
After performing some hyperparameter optimization, I want to train my selected model prototype on the entire available training set (i.e. train+val) without validating after epochs.

Is it possible to concatenate the two TabularDatsets somehow? 

ben0it9

**Comment by rumaak** [comment]
Yes, although it's not implemented directly in the library. My solution works with python `list`, but it doesn't seem slower than completely reloading the whole dataset. There are 2 similar solutions. In both cases, you need to extract list of `examples` from former datasets and create new one with these (meaning you will keep the former datasets). Then you just simply create new `Dataset` with those.

Minimal examples:
Common initialization
```
from torchtext.data import *
f1 = Field()
ds1 = TabularDataset(fname1,format,[('text',f1)])
ds2 = TabularDataset(fname2,format,[('text',f1)])
```
1 list way
```
ds_concat = ds1 + ds2
list_of_ex = [x for x in ds_concat]
new_ds = Dataset(list_of_ex,[('text',f1)])
```
2 lists way
```
list_of_ex1 = [x for x in ds1]
list_of_ex2 = [x for x in ds2]
list_of_ex = list_of_ex1 + list_of_ex2
new_ds = Dataset(list_of_ex,[('text',f1)])
```
The `new_ds` should be complete fully working datasets. These things worked for me:
```
f1.build_vocab(new_ds)
new_iter = Iterator(new_ds,batch_size)
next(iter(new_iter)).text
```
If anyone finds mistake in my approach / finds better one, I am open to suggestions.

**Comment by madaanpulkit** [comment]
I have multilingual parallel data. Therefore each of the individual (src-target) datasets has different fields because of different tokenisations(of different sources and targets). Is there a way to create a torchtext dataset for multilingual?

**Comment by nikitajz** [comment]
I've been looking for an answer to this question and didn't find it so implemented it myself. In my case, I don't need to concatenate datasets themselves, but I just want to iterate over the examples with the same fields* from different datasets hence I concatenated iterators (dataloaders). This is actually quite straightforward due to the simplicity of python's interface.

```
class MultiIterator:
    def __init__(self, iter_list):
        """MultiIterator to chain multiple iterators into one.

        Parameters
        ----------
        iter_list : [list]
            Sequence of torchtext iterators.
        """
        self.iters = iter_list

    def __iter__(self):
        for it in self.iters:
            for batch in it:
                yield batch

    def __len__(self):
        return sum(len(it) for it in self.iters)
```

Embedding vectors/vocabs should be handled separately but this is another story.

*by field here I mean the same name/type, not instance of class Field.

**Comment by abhinavarora** [comment]
Closing issue as we have gotten rid of`TabularDataset` and the issue is no more relevant. 
#1537
